### PR TITLE
S2-ONE-109 Edit profile

### PR DESCRIPTION
**Pull request by iwannabaa**
## Story / Task

[ONE-109](https://bixlabs.atlassian.net/browse/ONE-109)

## What this PR does?

Adds functionality to edit my profile information

### Before 
Could not edit ptofile info

### After
User is able to update their information

### How to test / reproduce

- Login to the app
- Using the mocked users data from previuos PRs connect to json-server (this is temporary solution, until we can use the API)
- Go to my profile page
- In the sidebar click, "Edit"
- Inputs for updating the info should appear
- Change data
- Click save

### TODOs
- [x] Rebase

### Related PRs
N/A

## Migrations

NO

### CODE REVIEW

File: src/app/profile/profile.service.ts [hunk 1]
```diff
@@ -27,4 +27,13 @@ export class ProfileService {
...
                .catch((error) => Observable.throw('Error'));
   }
+
+  saveProfile(profile: User) {
+    const url = `${CONFIG.api.url}/core/users/${profile.id}`;
+
+    return this.http
+               .patch(url, profile)
+               .map(response => <User>response.json())
+               .catch(error => Observable.throw('Error'));
```
Comment (evasquez26) [inline src/app/profile/profile.service.ts]: Use the exception service to handle any catch error

File: src/app/profile-sidebar/profile-sidebar.component.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import { Component, OnInit, Input } from '@angular/core';
 import { ActivatedRoute, Params }   from '@angular/router';
+import * as angular2ClickOutside from 'angular2-click-outside';
```
Comment (evasquez26) [inline src/app/profile-sidebar/profile-sidebar.component.ts]: This package is missing :)

Comment (iwannabaa) [inline src/app/profile-sidebar/profile-sidebar.component.ts]: Thanks. Missed that. I didn't use it in the end

File: src/app/profile/profile.service.ts (hunk 1 above)
Comment (iwannabaa) [inline src/app/profile/profile.service.ts]: Will update

**Pull request merged**